docs/recipes/reducers/00-PrerequisiteConcepts.md

```diff
@@ -0,0 +1,88 @@
+# Prerequisite Reducer Concepts
+
+
+As described in [Reducers](../../basics/Reducers.md), a Redux reducer function:
+
+- Should have a signature of `(previousState, action) => newState`, similar to the type of function you would pass to [`Array.prototype.reduce(reducer, ?initialValue)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce)
+- Should be "pure", which means it does not mutate its arguments, perform side effects like API calls or modifying values outside of the function, or call non-pure functions like `Date.now()` or `Math.random()`.  This also means that updates should be done in an ***"immutable"*** fashion, which means **always returning new objects with the updated data**, rather than directly modifying the original state tree in-place.
+
+>##### Note on immutability, side effects, and mutation
+> Mutation is discouraged because it generally breaks time-travel debugging, and React Redux's `connect` function.  For time traveling, the Redux DevTools expect that replaying recorded actions would output a state value, but not change anything else.  For React Redux, `connect` checks to see if the values returned from a `mapStateToProps` function have changed in order to see if a component needs to update.  Direct mutation can cause both of these scenarios to not work correctly.  Other side effects like generating unique IDs or timestamps in a reducer also make the code unpredictable and harder to debug.  
```

Comment:
> For React Redux, `connect` checks to see if the values returned from a `mapStateToProps` function have changed in order to see if a component needs to update

I wonder if we should clarify what we mean by 'changed' here? I've taken a stab at it below (also added a sentence giving an example of what may go wrong with DevTools).

>For time traveling, Redux DevTools expects that replaying recorded actions would output a state value, but not change anything else. Direct mutation of state may cause an app to end up in different final states each time a set of recorded actions is replayed. For React Redux, `connect` checks to see if a new value has been returned from a `mapStateToProps` function in order to determine if a component needs to update. `connect` **does not** use a deep equality check for this, so changes made to objects and arrays by direct mutation will not be detected. Other side effects like generating unique IDs or timestamps in a reducer also make the code unpredictable and harder to debug.